infrastructure/src/main/kotlin/extension/RedmadrobotExtension.kt

```diff
@@ -119,3 +127,32 @@ public class TestOptions {
         useJunitPlatform = false
     }
 }
+
+public class DetektOptions {
+
+    /** Options for detektDiff task */
+    internal var detektDiffOptions: DetektDiffOptions? = null
+        private set
+
+    /** Enable Detekt checks only for modified files provided by git (compare with [branch]) */
+    public fun checkOnlyDiffWithBranch(
+        branch: String,
+        configure: DetektDiffOptions.() -> Unit = {},
+    ) {
+        require(branch.isNotBlank()) { "Base branch not provided" }
```

Comment:
Message `"Base branch should not be blank."` will be more clear

Reply:
Fixed.